e2e/native-esm/__tests__/native-esm.test.js

```diff
@@ -72,7 +72,7 @@ test('import cjs', async () => {
   expect(half(4)).toBe(2);
 });
 
-test('import esm from cjs', async () => {
+test.skip('import esm from cjs', async () => {
```

Comment:
note that this makes the change unacceptable - this needs to work. Since https://github.com/nodejs/node/issues/31860 has been fixed, I assume you can just add the missing `importModuleDynamically` to the `compileFunction` call

Reply:
fyi - when I had left `importModuleDynamically` implemented, the memory leak persisted.  Another mystery...

Reply:
interesting! that might be a separate bug in node 😅

Reply:
I've replicated this myself locally. I'll see if I can figure out why.

Reply:
> I've replicated this myself locally. I'll see if I can figure out why.

Any luck @vanstinator ?

Reply:
@rthreei, the memory leak that occurs when using `compileFunction()` with `importModuleDynamically` is tracked [here](https://github.com/nodejs/node/issues/44211). A fix of that was attempted with [this PR](https://github.com/nodejs/node/pull/46785) a few weeks ago, but had to be reverted again.

Reply:
@rene-leanix Furthermore, it appears progression on this issue may be dependent upon this [V8 PR](https://chromium-review.googlesource.com/c/v8/v8/+/3172764) via this [comment](https://github.com/nodejs/node/issues/44211#issuecomment-1441980661)